addons/debugger/userscript.js

```diff
@@ -0,0 +1,296 @@
+export default async function ({ addon, global, console, msg }) {
+  let workspace, showingConsole;
+
+  const container = document.createElement("div");
+  container.className = "sa-debugger-container";
+  const buttonContainer = document.createElement("div");
+  buttonContainer.className = addon.tab.scratchClass("button_outlined-button", "stage-header_stage-button");
+  const buttonContent = document.createElement("div");
+  buttonContent.className = addon.tab.scratchClass("button_content");
+  const buttonImage = document.createElement("img");
+  buttonImage.className = addon.tab.scratchClass("stage-header_stage-button-icon");
+  buttonImage.draggable = false;
+  buttonImage.src = addon.self.dir + "/debug.svg";
+  buttonContent.appendChild(buttonImage);
+  buttonContainer.appendChild(buttonContent);
+  container.appendChild(buttonContainer);
+  buttonContainer.addEventListener("click", () => toggleConsole(true));
+
+  const vm = addon.tab.traps.vm;
+  addon.tab.addBlock("sa-log %s", ["content"], ({ content }, thread) => {
+    workspace = Blockly.getMainWorkspace();
```

Comment:
Why do we re-assign this on every call? Can't we do this downstream, when needed?